pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -1195,37 +1201,17 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
 		backup.Status.LastUpdateTimestamp = metav1.Now()
 		// temporarily store the volume status, So that it will be used during retry.
 		volumeInfos := backup.Status.Volumes
-		// Update the current state and then move on to backing up resources
-		err := a.client.Update(context.TODO(), backup)
+		backup, err = a.updateBackupCRInVolumeStage(
+			namespacedName,
+			stork_api.ApplicationBackupStatusInProgress,
+			backup.Status.Stage,
+			"Application resources backup is in progress",
+			volumeInfos,
+			backup.Status.FailedVolCount,
+		)
 		if err != nil {
-			for i := 0; i < maxRetry; i++ {
-				err = a.client.Get(context.TODO(), namespacedName, backup)
-				if err != nil {
-					time.Sleep(retrySleep)
-					continue
-				}
-				if backup.Status.Stage == stork_api.ApplicationBackupStageFinal ||
-					backup.Status.Stage == stork_api.ApplicationBackupStageApplications {
-					log.ApplicationBackupLog(backup).Infof("Returning from ApplicationBackupStageVolumes since the CR is in %v stage", backup.Status.Stage)
-					log.ApplicationBackupLog(backup).Warnf("post-exec rule can get called more than once in case of kdmp & mixed driver scenario, please check logs for the same")
-					return nil
-				}
-				backup.Status.Stage = stork_api.ApplicationBackupStageApplications
-				backup.Status.Status = stork_api.ApplicationBackupStatusInProgress
-				backup.Status.Reason = "Application resources backup is in progress"
-				backup.Status.LastUpdateTimestamp = metav1.Now()
-				backup.Status.Volumes = volumeInfos
-				err = a.client.Update(context.TODO(), backup)
-				if err != nil {
-					time.Sleep(retrySleep)
-					continue
-				} else {
-					break
-				}
-			}
-			if err != nil {
-				return err
-			}
+			logrus.Errorf("Error while updateBackupCRInVolumeStage: %v", err)
```

Comment:
Nit - > can log.ApplicationBackupLog(backup).Errorf used instead?

Reply:
It can be used, at a lot of places the use is a bit inconsistent. Hence, going with this.